### PR TITLE
Moving get_default_solver method

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This is the first part of addressing #202. This moves the existing `get_default_solver` method from util.testing to util.misc to avoid a potential recursive import loop and updates all existing models and tests that rely on the method. This will provide a centralized method for creating a default solver object in the absence of user input. For now, the solver options are hard-coded, but we can look at moving to using the IDAES config file in future (and all models/tests that use the method will automatically be updated0.

## Changes proposed in this PR:
- Moved the `get_default_solver` method to `util.misc`, with a wrapper method left in `util.testing` with a deprecation warning for backwards compatibility.
- Updated the default initialize methods in `UnitBlockData` to use the new `get_default_solver` method in place of a hard-coded iopt instance.
- Changed the default values of the `solver` and `optarg` arguments in the `ControlVolume0D` and `ControlVolume1D` initialize methods to not presume any settings.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/util/misc.py [hunk 1]
```diff
@@ -29,6 +29,22 @@
...
+# Author: Andrew Lee
+def get_default_solver():
+    """
+    Tries to set-up the default solver for testing, and returns None if not
+    available
+    """
+    if pyo.SolverFactory('ipopt').available(exception_flag=False):
+        solver = pyo.SolverFactory('ipopt')
+        solver.options = {'tol': 1e-6,
+                          'linear_solver': 'ma27'}
```
Comment (eslickj) [inline idaes/core/util/misc.py:41]: If you are using Ipopt and ma27 is available, I believe ma27 is the default linear solver.  So the IDAES version should default to ma27.  Specifying this option causes the solve to fail if ma27 isn't available, which I guess could be what you are after.  If someone has Ipopt and no ma27 it is probably a small number of failures if you don't specify this option vs everything failing if you do.

File: idaes/core/util/misc.py [hunk 2]
```diff
@@ -29,6 +29,22 @@
...
 
 
+# Author: Andrew Lee
+def get_default_solver():
+    """
+    Tries to set-up the default solver for testing, and returns None if not
+    available
+    """
+    if pyo.SolverFactory('ipopt').available(exception_flag=False):
+        solver = pyo.SolverFactory('ipopt')
```
Comment (eslickj) [inline idaes/core/util/misc.py:39]: It doesn't really mater, but we'll probably want to standardize on "ipopt-idaes" here.  The default settings will probably be the same as plain ipopt forever, but just for future consistency and to set a good example.

**Comment by andrewlee94** [comment]
@eslickj My intent is to standardize the solver and options in another PR. For now, I thought this was getting big enough with just moving things around, so I put it in first.

**Review by eslickj — APPROVED** [review]

**Review by MAZamarripa — APPROVED** [review]
lgtm

**Pull request merged**